### PR TITLE
feat: filter thinking tags from copied assistant responses

### LINKED ISSUE #118: Don't copy thinking tags
When copying assistant response, ignore the text in the think tags

### PR DESCRIPTION
**Pull request by AnthonyRonning**
**Summary**

- Implemented thinking tag filtering for copy functionality
- Added stripThinkingTags utility function to remove <think>...</think> tags
- Updated SystemMessage copy button to use filtered content

**Test plan**

- [ ] Test copying assistant responses with thinking tags
- [ ] Verify thinking tags are excluded from copied text
- [ ] Test copying responses without thinking tags still works
- [ ] Test incomplete thinking tags are handled correctly

Fixes #118

🤖 Generated with [Claude Code](https://claude.ai/code)

## Summary by CodeRabbit

- **New Features**
  - Improved copy functionality in chat messages by automatically removing hidden thinking tags from copied text for a cleaner result.

### CODE REVIEW

File: frontend/src/components/markdown.tsx [hunk 1]
```diff
@@ -176,6 +176,22 @@ function parseThinkingTags(content: string, isComplete: boolean = false): Parsed
   return parts;
 }
 
+export function stripThinkingTags(content: string): string {
+  // Handle the edge case: <think> followed by only whitespace and no closing tag
+  if (/^<think>\s*$/m.test(content) && !content.includes("</think>")) {
+    return content.replace(/^<think>\s*/m, "");
+  }
```
[the remark this answers is not in the document]
Comment (AnthonyRonning) [reply frontend/src/components/markdown.tsx:183]: @claude handle some of the duplicate logic we have based on the PR review comments we have here.
